### PR TITLE
Adds initial state of eu workshop

### PR DESCRIPTION
**Pull request by burtlo**
The scenario contains one working exercise for the apps client.

The remaining two exercises need to be updated.

- [x] step-app-2 updated (@burtlo)
- [x] step-app-2-answer updated (@burtlo)
- [x] step-app-3 updated (@burtlo)
- [x] step-app-3-answer updated (@burtlo)

The remaining exercises need to be created.

- [x] step-app-4 focuses on using `*` and `+`. (@burtlo)
- [x] step-app-4-answer (@burtlo)
- [ ] step-app-5 focuses on using path templating (@burtlo)
- [ ] step-app-5-answer (@burtlo)

The `admin` client/persona is the next set of exercises to create.

- [x] step-admin-intro describes this client and displays their policy (@burtlo)
- [x] step-admin-1 focuses on kv-v2 secret management (@burtlo)
- [x] step-admin-1-answer (@burtlo)
- [x] step-admin-2 focuses on database secret management (@burtlo)
- [x] step-admin-2-answer (@burtlo)
- [x] step-admin-3 focuses on transit secret management (@burtlo)
- [x] step-admin-3-answer (@burtlo)

The current answers for admin step 2 and admin step 3 contain the use of `+` and `*`. Those could be revised to be more verbose and then they are refactored in step 4. Or they could be better explained.

- [ ] step-admin-4 focuses on using the `*` and `+`
- [ ] step-admin-4-answer
- [ ] step-admin-5 focuses on using path templating
- [ ] step-admin-5-answer

### CODE REVIEW

File: vault-granular-access-control/step-app-1.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Select the KV-V2 API tab to view the [KV-V2 API
+documentation](https://www.vaultproject.io/api-docs/secret/kv/kv-v2).
+
+The [read secret
+version](https://www.vaultproject.io/api-docs/secret/kv/kv-v2#read-secret-version)
+operation describes the capability and the path. The operation requires the
+`GET` HTTP verb which translates to the `read` capability. The templatized path
+`/secret/data/:path` becomes `/socials/data/twitter` when the path element
+provided by the secret.
```
Comment (brianshumate) [inline vault-granular-access-control/step-app-1.md:102]:
```suggestion
is provided by the secret.
```

File: vault-granular-access-control/step-app-2.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+```
+               __
```
Comment (brianshumate) [inline vault-granular-access-control/step-app-2.md:2]: The ASCII art here is great!

File: vault-granular-access-control/step-app-2.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    ..=====.. |==|      ___________________
+    ||     || |= |     < Database, please! >
+ _  ||     || |^*| _    -------------------
+|=| o=,===,=o |__||=|
+|_|  _______)~`)  |_|
+    [=======]  ()
+```
+
+The web application needs access to database
```
Comment (brianshumate) [inline vault-granular-access-control/step-app-2.md:11]:
```suggestion
The web application needs access to a database.
```

File: vault-granular-access-control/step-app-2.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+```shell
+vault login -method=userpass \
+  username=apps \
+  password=apps-password
+```{{execute}}
+
+## Perform action
+
+Attempt to read credentials from the database role
```
Comment (brianshumate) [inline vault-granular-access-control/step-app-2.md:23]:
```suggestion
Attempt to read credentials from the database role.
```

File: vault-granular-access-control/step-app-3.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+               __
+    ..=====.. |==|      _____________________
+    ||     || |= |     < Encryption, please! >
+ _  ||     || |^*| _    ---------------------
+|=| o=,===,=o |__||=|
+|_|  _______)~`)  |_|
+    [=======]  ()
+```
+
+The web application needs access to the transit encryption key
```
Comment (brianshumate) [inline vault-granular-access-control/step-app-3.md:11]:
```suggestion
The web application needs access to the transit encryption key.
```

File: vault-granular-access-control/step-app-3.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+```shell
+vault login -method=userpass \
+  username=apps \
+  password=apps-password
+```{{execute}}
+
+## Perform action
+
+Attempt to encrypt content with transit key
```
Comment (brianshumate) [inline vault-granular-access-control/step-app-3.md:23]:
```suggestion
Attempt to encrypt content with transit key.
```

File: vault-granular-access-control/step0.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+```{{execute}}
+
+
+Display the status of the target Vault server.
+
+```shell
+vault status
+```{{execute}}
+
+The response shows that the Vautl server is initialized and unsealed.
```
Comment (brianshumate) [inline vault-granular-access-control/step0.md:19]:
```suggestion
The response shows that the Vault server is initialized and unsealed.
```

**Review by brianshumate — CHANGES_REQUESTED** [review]
Fantastic scenario! I am inspired and will definitely be borrowing heavily from this for the Sentinel Simulator scenario. Love the ASCII art, it's such a cool and humanistic touch. Just suggested a few grammar things thus far, but otherwise this is awesome!

File: vault-granular-access-control/step-app-1.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+```shell
+vault kv get -output-curl-string socials/twitter
+```
+
+curl -H "X-Vault-Request: true" -H "X-Vault-Token: $(vault print token)"
+http://localhost:8200/v1/socials/data/twitter
+
+The HTTP verb by default is `GET` which translates to the `read` capability.
+The requested URL displays the path `/socials/data/twitter`.
+
```
[the remark this answers is not in the document]
Comment (burtlo) [reply vault-granular-access-control/step-app-1.md:71]: I definitely think this explanation could more thorough.

**Comment by burtlo** [comment]
#ad16c40

- fixes layout of ascii art in smaller browser window
- updates to postgres setup as `ro` creation was failing
- updates to the vault server to log in trace
- updates to the app requirement #1 key to potentially support step 4
- step 1 uses the `output-curl-string`
- delivers step 2
- step 2 mentions #1 and then introduces audit log
- delivers step 3
- step 3 mentions #1 and #2, then introduces api docs

Future applications in this series will rely on these techniques and
only use short mention of that approach.

File: vault-granular-access-control/index.json [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+        {
+          "file": "apps-policy.hcl",
+          "target": "~/"
+        }
+      ]
+    }
+  },
+  "environment": {
+      "showdashboard": true,
+      "uilayout": "terminal",
```
Comment (yhyakuna) [inline vault-granular-access-control/index.json:85]: Maybe use Katacoda text editor to make it easier to write policies???

```suggestion
      "uilayout": "editor-terminal",
```

File: vault-granular-access-control/step-admin-1-answer.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+path \"external-apis/data/socials/+\" {
+  capabilities = [ \"create\", \"read\", \"update\", \"delete\" ]
+}
+
+path \"external-apis/undelete/socials/+\" {
+  capabilities = [ \"update\" ]
+}
+" >> admins-policy.hcl
+```{{execute}}
```
Comment (yhyakuna) [inline vault-granular-access-control/step-admin-1-answer.md:35]: If we use the `editor-terminal` layout, you can do...

```suggestion
Open the `admins-policy.hcl`{{open}} and append the following policies.

<pre class="file" data-filename="admins-policy.hcl" data-target="append">
path "external-apis/data/socials/twitter" {
  capabilities = [ "create", "read", "update", "delete" ]
}
path "external-apis/data/socials/+" {
  capabilities = [ "create", "read", "update", "delete" ]
}
path "external-apis/undelete/socials/+" {
  capabilities = [ "update" ]
}
</pre>
```

That's the trick I use in the [Raft HA Storage](https://www.katacoda.com/hashicorp/scenarios/vault-raft-ha) tutorial.

File: vault-granular-access-control/index.json (hunk 9 above)
Comment (burtlo) [reply vault-granular-access-control/index.json:85]: I tend to hate how much the file editor takes up but this the right move for ease-of-use.

**Comment by burtlo** [comment]
Updated all the answer steps to use that approach. Thank you @yhyakuna.

**Comment by burtlo** [comment]
Added App Step 4 with a rudimentary application of `+` operator. I would like for the requirement to include a second key but need to exclude a third key (Use of `+` instead of a `*`).

**Comment by yhyakuna** [comment]
@burtlo  I'm merging this so that I can schedule a training for Wednesday

**Pull request merged**